src/services/jsDoc.ts

```diff
@@ -246,6 +248,12 @@ export function getJsDocTagsFromDeclarations(declarations?: Declaration[], check
         }
         for (const tag of tags) {
             infos.push({ name: tag.tagName.text, text: getCommentDisplayParts(tag, checker) });
+
+            if (isJSDocPropertyLikeTag(tag) && tag.isNameFirst && tag.typeExpression && isJSDocTypeLiteral(tag.typeExpression.type)) {
+                forEach(tag.typeExpression.type.jsDocPropertyTags, propTag => {

```

Comment:
does this improve or change the way `@typedef` displays?
`@callback` and (I think) `@overload` have similar nested tag structure, too. Do they have similar problems displaying quickinfo? They might not since they all exist as their own comments.

Reply:
@sandersn I think this should not impact tags that have their own comments. Does the test [jsDocTypedefQuickInfo1](https://github.com/microsoft/TypeScript/blob/main/tests/cases/fourslash/jsDocTypedefQuickInfo1.ts) cover the display aspects for the typedef? Should I expand it to include additional cases?

Reply:
That test doesn't really cover the expanded display; it doesn't display the tags for the typedef at all because it's requesting the type for a parameter.

On the other hand, quick info display for typedefs themselves is *decent* already -- but it doesn't show the individual tags. For example

```js
/**
  * @typedef {Object} Opts
  *                   ^^^
  * @property {string} x hello
  * @property {string=} y hup
  * @property {string} [z] hap
  * @property {string} [w="hi"] hip
  *
*/
```

Shows the TS syntax for `Opts`, which is useful, but it would be nice to also show the property tags themselves.
*However*, that's a different feature from this change, and I'd like to get this one in on its own.